### PR TITLE
Add support for ForkJoin Pool

### PR DESCRIPTION
**Pull request by aubi**
Add implementation of concurrency using JDK forkjoin pool instead of own thread pool.

### CODE REVIEW

File: src/main/java/org/glassfish/enterprise/concurrent/ForkJoinManagedExecutorService.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+// Portions Copyright [2022-2023] [Payara Foundation and/or its affiliates]
```
Comment (arjantijms) [inline src/main/java/org/glassfish/enterprise/concurrent/ForkJoinManagedExecutorService.java:16]: Portions Copyright is not specifically needed for the EPL 2.0. You can add a new copyright line to the top.

File: src/main/java/org/glassfish/enterprise/concurrent/ManagedThreadFactoryImpl.java [hunk 2]
```diff
@@ -187,20 +187,18 @@ protected Collection<Thread> getThreads() {
         return result;
     }
     public void taskStarting(Thread t, ManagedFutureTask task) {
-        if (t instanceof ManagedThread) {
-            ManagedThread mt = (ManagedThread) t;
+        if (t instanceof ThreadWithTiming) {
```
Comment (arjantijms) [inline src/main/java/org/glassfish/enterprise/concurrent/ManagedThreadFactoryImpl.java:190]: You can omit the cast and just write

`if (t instanceof ThreadWithTiming mt)`

Comment (aubi) [reply src/main/java/org/glassfish/enterprise/concurrent/ManagedThreadFactoryImpl.java]: I was using still JDK 11, but you are right, pom was upgraded to 17. I updated all such cases in this file and ForkJoinManagedExecutorService.

**Review by arjantijms — APPROVED** [review]

**Comment by arjantijms** [comment]
LGTM!

**Review by avpinchuk — APPROVED** [review]

**Pull request merged**